qadence/transpile/emulate.py

```diff
@@ -275,7 +275,13 @@ def _(block: ConstantAnalogRotation, register: Register, interaction: Callable)
     # convert "global" to indexed qubit suppport so that we can re-use `kron` dispatched function
     b = deepcopy(block)
     b.qubit_support = QubitSupport(*range(register.n_qubits))
-    return _add_interaction(kron(b), register, interaction)
+
+    # convert to HamEvo block
+    hamevo_block = _add_interaction(kron(b), register, interaction)
+
+    # set eigenvalues of the ConstantAnalogRotation block
+    block.eigenvalues_generator = hamevo_block.eigenvalues_generator
```

Comment:
This is a bit risky since you are modifying the block in place. It might give some nasty side effects? Should we maybe rethink a little bit the `add_interaction` function and its behavior? I guess this is something that @jpmoutinho will take care of in his refactoring anyway.

Reply:
You see, the problem here is that we need to have eigenvalues for the `ConstantAnalogRotation` block itself so this in-place modification is unavoidable with the current block implementation. This calculation of eigenvalues may be done in some different place but in the end you still have to assign them to the analog block itself.